### ISSUE TITLE
The hook to dired renamer fixes links incorrectly

### ISSUE THREAD

**Issue by jcguu95** [org-roam/org-roam#1041]
TL;DR: need a better renamer.

### Description

Renaming files under `org-roam` via `dired` invokes a `org-roam` hook that fixes the links in the whole system. I find this risky. Indeed there's a bug.

#### Steps to Reproduce

1. Setup

```
cd org-roam
mkdir testing; cd testing
mkdir deeper
```

Link `testing/a` and `testing/b` both in `a.org` and `b.org`.

2. Use `dired` to **batch** move `a.org` and `b.org` into `deeper`, by marking them by `m` and move by `R`.

3. Observe that `roam` fixes the link incorrectly.

#### Expected Results

In the end both `a.org` and `b.org` are in the same dir, so the links should not be changed.

#### Actual Results

```
cat a.org
[[file:../a.org][testing/a]]
[[file:../b.org][testing/b]]

cat b.org
[[file:a.org][testing/a]]
[[file:../b.org][testing/b]]
```

### Solution Idea

Create a safer "file renamer", and hook it to dired's renaming function. Make the file renamer callable for end users.

**Comment by kljohann** [comment]
Depending on how this is hooked into dired, `dired-vc-rename-file` (new in Emacs 27.1) could also be considered.

**Comment by jethrokuan** [comment]
Renames now only cause database changes in v2, not an issue anymore.
